lib/utils/report.mjs

```diff
@@ -93,14 +96,24 @@ export default function report(problem) {
 
 	if (start) {
 		warningProperties.start = start;
-	} else if (isNumber(index)) {
-		warningProperties.index = index;
+	} else if (range?.start) {
+		warningProperties.start = range.start;
+	} else if (isNumber(startLine)) {
+		warningProperties.start = {
+			line: startLine,
+			column: 1,
+		};
```

Comment:
It is impossible to have an `index` value that is accurate and not to have a `range.start`.

A rule could create a report with an `index` and `line` while omitting the `node`, but then the `index` can't be used anyway.

The `index` is relative to the `node`, and when the `node` is missing, the `index` is essentially bogus.

If a rule does create such a report it is better to completely ignore the `index` and use only the `line`.

Reply:
I don't like the fallback for `column`.
If it's actually better than `} else if (isNumber(index)) {` could you add a comment that explains that choice?

Reply:
> I don't like the fallback for `column`.

Setting the report to the first column on the given line seemed like an ok fallback to me.

We could calculate it by scanning the input line by line until we encounter the line in question. This seemed like overkill to me.

> If it's actually better than `} else if (isNumber(index)) {` could you add a comment that explains that choice?

_Already explained it in detail, but second attempt :)_

`index` is relative to the `node`. `index: 0` corresponds the first position in the node itself.

It is not the first position in the original source.

However where there is no `node` the `index` becomes meaningless.
PostCSS won't do anything with it and neither will anything else.

Its only purpose is to be used in a `node.range(...)` call.
If there is no node, there is no such call.

So `index` can't be used as a fallback for `range.start`.
Either there is a `node` and then there also is a `range.start` value or there is no `node` and then we can't use anyway.

Reply:
> This seemed like overkill to me.

👍 

> Already explained it in detail, but second attempt :)

Sorry maybe I wasn't clear, I meant: could you add a comment _in the code_ that explains why we are using magic numbers ? (i.e. `1` and `2`)
Your initial comment—_in the PR_—was very clear.

Reply:
I've added a comment, hopefully that is enough of a hint.
I didn't want to inline the entire explanation 😅